### PR TITLE
instructorStudentsList: open record in new window #1409

### LINKED ISSUE #1409: instructorStudentsList: open record in new window
_From [dam...@gmail.com](https://code.google.com/u/114919422028123491955/) on January 03, 2014 13:23:46_

Loading student list page is expensive. If we load the 'all records' page in a new window, we can minimize the need to reload the student list page.

_Original issue: http://code.google.com/p/teammatespes/issues/detail?id=1410_

### PR DESCRIPTION
**Pull request by github-tn**
Fixes #1409

### CODE REVIEW

File: src/test/java/teammates/test/pageobjects/AppPage.java [hunk 1]
```diff
@@ -295,6 +295,7 @@ public void reloadPage() {
     public <T extends AppPage> T goToPreviousPage(Class<T> typeOfPreviousPage) {
         browser.driver.navigate().back();
         waitForPageToLoad();
+        switchToNewWindow();
```
Comment (LowWeiLin) [inline src/test/java/teammates/test/pageobjects/AppPage.java]: Is this change required? It seems like the goToPreviousPage function might be called elsewhere, and that this statement will change the behavior.

File: src/test/resources/pages/NJIfeedbackSessionRemind.html [hunk 2]
```diff
@@ -385,7 +385,7 @@ <h5 class="col-sm-2">
                             <div data-original-title="Select this option to automatically send an email to students to notify them when the session is open for submission." class="col-sm-3" title="" data-toggle="tooltip" data-placement="top">
                                 <div class="checkbox">
                                     <label>Session opening
-                                        reminder</label> <input checked="checked" name="sendreminderemail" id="sendreminderemail_open" value="FEEDBACK_OPENING" type="checkbox" disabled="disabled">
+                                        reminder</label> <input checked="checked" name="sendreminderemail" id="sendreminderemail_open" value="FEEDBACK_OPENING" disabled="disabled" type="checkbox">
```
Comment (LowWeiLin) [inline src/test/resources/pages/NJIfeedbackSessionRemind.html]: This change does not seem related to this issue. Does it cause a test failure if unchanged?

File: src/test/resources/pages/instructorStudentListPageWithPicture.html [hunk 3]
```diff
@@ -154,7 +154,7 @@
...
...oggleDeleteStudentConfirmation('course3','Denny Charlés')"> Delete</a>&nbsp;<a data-original-title="View all student's evaluations and feedbacks" class="btn btn-default btn-xs student-records-for-test" href="/page/instructorStudentRecordsPage?courseid=course3&amp;studentemail=denny.c.tmms@gmail.tmt&amp;user=instructorWith2Courses" title="" target="_blank" data-toggle="tooltip" data-placement="top"> All Records</a>&nbsp;<div class="dropdown inline"><a class="btn btn-default btn-xs dropdown-toggle" href="javascript:;" data-toggle="dropdown"> Add Comment</a><ul class="dropdown-menu align-left" role="menu" aria-labelledby="dLabel"><li role="presentation"><a target="_blank" role="menuitem" tabindex="-1" href="/page/instructorCourseStudentDetailsPage?courseid=course3&amp;studentemail=denny.c.tmms@gmail.tmt&amp;user=instructorWith2Courses&amp;addComment=student">Comment on Denny Charlés</a></li><li role="presentation"><a target="_blank" role="menuitem" tabindex="-1" href="/page/instructorCourseStudentDetailsPage?courseid=course3&amp;studentemail=denny.c.tmms@gmail.tmt&amp;user=instructorWith2Courses&amp;addComment=team">Comment on Team 2</a></li></ul></div></td></tr></tbody></table></div>
```
Comment (LowWeiLin) [inline src/test/resources/pages/instructorStudentListPageWithPicture.html]: This test is failing. There is a pixel value that need to be change to wildcard "{*}px", as different browser size will cause the pixel value to change. search for "px" to find it.

Comment (github-tn) [inline src/test/resources/pages/instructorStudentListPageWithPicture.html]: okay I will fix this.

File: src/test/java/teammates/test/pageobjects/AppPage.java (hunk 1 above)
Comment (github-tn) [inline src/test/java/teammates/test/pageobjects/AppPage.java]: I think no need to add this line here. Forgot why I added it here...

File: src/test/resources/pages/NJIfeedbackSessionRemind.html (hunk 2 above)
Comment (github-tn) [inline src/test/resources/pages/NJIfeedbackSessionRemind.html]: I revert this change and passed all test cases.

**Comment by LowWeiLin** [comment]
@damithc Looks good to me.
@github-tn You can set the label to s.ToReview when you think the pull request is ready for review, and s.Ongoing when making changes. Refer to the process document for details.

**Comment by damithc** [comment]
Nice. Go ahead and merge @LowWeiLin
